nvflare/lighter/nvflare.py

```diff
@@ -17,45 +17,58 @@
 import sys
 
 from nvflare.lighter.poc_commands import def_poc_parser, handle_poc_cmd, is_poc
+from nvflare.lighter.provision import define_provision_parser, handle_provision
+from nvflare.tool.preflight_check import check_packages, define_preflight_check_parser
 
 
-def handle_provision_cmd(cmd_args):
-    print("handle provision command")
-    pass
+def is_provision(cmd_args) -> bool:
+    return (
+        hasattr(cmd_args, "add_user")
+        or hasattr(cmd_args, "add_client")
+        or hasattr(cmd_args, "project_file")
+        or hasattr(cmd_args, "ui_tool")
+    )
 
 
-#
-#
-# def is_provision(cmd_args) -> bool:
-#     #  todo add provision handling
-#     return False
-#
-#
-# def def_provision_parser(sub_cmd, prog_name: str):
-#     provision_parser = sub_cmd.add_parser("provision")
-#     provision_parser.add_argument(
-#         "-n", "--n_clients", type=int, nargs="?", default=2, help="number of sites or clients"
-#     )
+def is_preflight_checker(cmd_args) -> bool:
+    print(cmd_args)
+    return hasattr(cmd_args, "package_root") or hasattr(cmd_args, "packages]")
+
+
+def def_provision_parser(sub_cmd):
+    provision_parser = sub_cmd.add_parser("provision")
+    define_provision_parser(provision_parser)
+
+
+def def_preflight_check_parser(sub_cmd):
+    checker_parser = sub_cmd.add_parser("preflight_check")
+    define_preflight_check_parser(checker_parser)
 
 
 def parse_args(prog_name: str):
-    _parser = argparse.ArgumentParser(description="nvflare parser")
+    _parser = argparse.ArgumentParser(description=prog_name)
     sub_cmd = _parser.add_subparsers(description="sub command parser")
-    def_poc_parser(sub_cmd, prog_name)
-    # def_provision_parser(sub_cmd, prog_name)
+    def_poc_parser(sub_cmd)
+    def_provision_parser(sub_cmd)
+    def_preflight_check_parser(sub_cmd)
     return _parser, _parser.parse_args()
 
 
 def run(prog_name):
     cwd = os.getcwd()
     sys.path.append(cwd)
     prog_parser, prog_args = parse_args(prog_name)
-
-    if is_poc(prog_args):
-        handle_poc_cmd(prog_args)
-    # elif is_provision(prog_args):
-    #     handle_provision_cmd(prog_args)
-    else:
+    try:
+        if is_poc(prog_args):
+            handle_poc_cmd(prog_args)
+        elif is_provision(prog_args):
+            handle_provision(prog_args)
+        elif is_preflight_checker(prog_args):
+            check_packages(prog_args)
+        else:
+            prog_parser.print_help()
+    except Exception as e:
+        print("unable to handle command, please check syntax")
         prog_parser.print_help()
```

Comment:
if we have this generic entry "nvflare cli" inside lighter, maybe we just move all the preflight check codes into the "nvflare/lighter" folder?

Reply:
but that can be done in a separate PR.